### PR TITLE
Resolve version constraints for black package installation

### PR DESCRIPTION
**Pull request by Akshat2Jain**
### This pull request introduces several updates and improvements to the codebase. Here is a summary of the changes:

- Created a new constraints.txt file.
- Updated the Makefile and CI configuration file to use the constraints.txt file for installing black.
- Modified the CI workflow to include the installation of black using the -c constraints.txt flag.

This Pr is related to the #1668

### CODE REVIEW

**Comment by Akshat2Jain** [comment]
@wesleybl, Can you review it?

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -24,6 +24,10 @@ jobs:
       - name: install black
         run: pip install click==$(awk '/^click =/{print $NF}' versions.cfg) black==$(awk '/^black =/{print $NF}' versions.cfg)
```
Comment (wesleybl) [inline .github/workflows/black.yml:25]: You must replace these lines. Not add others.

**Review by wesleybl — CHANGES_REQUESTED** [review]
The black CI is failing:

https://github.com/plone/plone.restapi/pull/1671#partial-pull-merging

**Comment by wesleybl** [comment]
Please remove [versions.cfg](https://github.com/plone/plone.restapi/blob/master/versions.cfg) file.

**Comment by Akshat2Jain** [comment]
@wesleybl, Can you check constraints.txt once, is there something to add?

**Comment by wesleybl** [comment]
@Akshat2Jain the error gives you a hint. Click `Details` on the black error in: https://github.com/plone/plone.restapi/pull/1671#partial-pull-merging

**Comment by Akshat2Jain** [comment]
Flake 8 is failing, Do I have to do the same as black

**Review by wesleybl — APPROVED** [review]
LGTM

**Review by davisagli — APPROVED** [review]
Thanks!

**Comment by polyester** [comment]
@wesleybl sure, no problem and done.

**Comment by Akshat2Jain** [comment]
@wesleybl @polyester @fredvd Thank You, guys!

**Pull request merged**